src/settings.js

```diff
@@ -47,11 +47,14 @@ var INVOKE_RUN = 1; // Whether we will run the main() function. Disable if you e
 var NO_EXIT_RUNTIME = 0; // If set, the runtime is not quit when main() completes (allowing code to
                          // run afterwards, for example from the browser main event loop).
 var MEM_INIT_METHOD = 0; // How to represent the initial memory content.
-                         // 0: keep array literal representing the initial memory data
+                         // 0: embed a base64 string literal representing the initial memory data
                          // 1: create a *.mem file containing the binary data of the initial memory;
                          //    use the --memory-init-file command line switch to select this method
                          // 2: embed a string literal representing that initial memory data
-                         //    XXX this is known to have bugs on windows, see https://github.com/kripken/emscripten/pull/3326
+                         //    XXX this is known to have bugs on windows, see https://github.com/kripken/emscripten/pull/3326;
+                         //    due to mode 0's recent upgrade to space-efficient base64 encoding
+                         //    and the long-standing issues with this mode, this mode is now
+                         //    deprecated in favor of mode 0
```

Comment:
What is this deprecation about?

In current `incoming` branch without this PR applied, running `emcc tests/hello_world.c -o hello_world.html -s MEM_INIT_METHOD=2` results in a .js file that is 294,149 bytes. In this PR branch, running `emcc tests/hello_world.c -o hello_world.html -s SINGLE_FILE=1` yields a .js file that is 315,541 bytes (+7.2% larger).

I think that `MEM_INIT_METHOD=2` is more efficient than base64 encoding also in general, not just on hello world. Also, in current `incoming` before this PR, `emcc tests/hello_world.c -o hello_world.html -s MEM_INIT_METHOD=0` results in a .js file of 299,555 bytes, but in this PR, same command gives 315,541 bytes (+5.3% larger). It could be that some of that is due to unrelated changes in between this PR has been worked on, so it could be good to rebase against latest incoming to compare more precisely.